pkg/cloudprovider/gce/gce.go

```diff
@@ -154,6 +178,7 @@ func newGCECloud(config io.Reader) (*GCECloud, error) {
 		projectID:        projectID,
 		zone:             zone,
 		instanceID:       instanceID,
+		networkName:      networkName,
```

Comment:
We're actively eliminating all cases where the master contacts the nodes for management purposes.

The only remaining cases should be the "proxy" functionality: proxy, redirect, port forwarding, exec, logs. That functionality will eventually be factored out into a separate component that could run in the same network as the kubelets, and the apiserver could proxy/redirect to it via a public IP.